src/lib/operators/aggregate_sort.cpp

```diff
@@ -368,14 +431,55 @@ std::shared_ptr<const Table> AggregateSort::_on_execute() {
    * However, we did not benchmark it, so we cannot prove it.
    */
 
-  // Sort input table consecutively by the group by columns (stable sort)
-  auto sorted_table = input_table;
-  for (const auto& column_id : _groupby_column_ids) {
-    const auto sorted_wrapper = std::make_shared<TableWrapper>(sorted_table);
-    sorted_wrapper->execute();
-    Sort sort = Sort(sorted_wrapper, std::vector<SortColumnDefinition>{SortColumnDefinition{column_id}});
-    sort.execute();
-    sorted_table = sort.get_output();
+  auto table_value_clustered_by = input_table->value_clustered_by();
+
+  std::shared_ptr<const Table> sorted_table = input_table;
+  if (!_groupby_column_ids.empty()) {
+    /**
+    * If there is a value clustering for a column, it means that all tuples with the same value in that column
+    * are in the same chunk.
+    * Therefore, if one of the value clustering columns is part of the group by vector, we can skip
+    * sorting the whole table and sort on a per-chunk basis instead. Values that would end up in the same cluster
+    * are already in the same chunk.
+    *
+    * If the value clustering doesn't match the columns we group by, we need to re-sort the whole table
+    * to achieve the value clustering we need (table-wide sort information is not tracked yet).
+    */
+
+    auto is_value_clustered_by_groupby_column = false;
+
+    if (table_value_clustered_by) {
+      for (const auto& value_clustered_by : *table_value_clustered_by) {
+        if (std::find(_groupby_column_ids.begin(), _groupby_column_ids.end(), value_clustered_by) !=
+            _groupby_column_ids.end()) {
+          is_value_clustered_by_groupby_column = true;
+          break;
+        }
+      }
+
+      if (is_value_clustered_by_groupby_column) {
+        // Sort input table chunk-wise consecutively by the group by columns (stable sort)
+        sorted_table = _sort_within_chunks(input_table, _groupby_column_ids);
+      }
+    }
+    if (!table_value_clustered_by || !is_value_clustered_by_groupby_column) {
+      // sort input table in whole consecutively by the group by columns
+      for (const auto& column_id : _groupby_column_ids) {
+        auto table_wrapper = std::make_shared<TableWrapper>(sorted_table);
+        table_wrapper->execute();
+        auto sort =
+            std::make_shared<Sort>(table_wrapper, std::vector<SortColumnDefinition>{SortColumnDefinition{column_id}});
+        sort->execute();
+        sorted_table = sort->get_output();
+      }
+      auto last_chunk = sorted_table->last_chunk();
+      if (last_chunk->is_mutable()) {
+        last_chunk->finalize();
+      }
```

Comment:
?

Reply:
These look like remnants from using the previous sort operator. Will take a look at it.

Reply:
Removed.